onnxruntime/test/python/onnxruntime_test_collective.py

```diff
@@ -85,6 +85,24 @@ def test_all_gather(self):
 
         assert np.allclose(outputs[0], expected_output)
 
+    def test_all_gather_axis1(self):
+        model = self._create_allgather_ut_model((128, 128), 1)
+        rank, size = self._get_rank_size()
+        ort_sess = ort.InferenceSession(
+            model.SerializeToString(),
+            providers=["CUDAExecutionProvider", "CPUExecutionProvider"],
+            provider_options=[{"device_id": str(rank)}, {}],
+        )
+
+        input = np.ones((128, 128), dtype=np.float32) * rank
+        outputs = ort_sess.run(None, {"X": input})
+
+        expected_output = np.zeros((128, 128), dtype=np.float32)
+        for _ in range(size - 1):
+            expected_output = np.concatenate((expected_output, np.ones((128, 128), dtype=np.float32) * (_ + 1)), axis=1)
+
+        assert np.allclose(outputs[0], expected_output)
```

Comment:
Consider `np.testing.assert_allclose` for a more detailed message when there is an error?